app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -286,36 +293,44 @@ class SubscriptionController @Inject()(
   }
 
   val showLinkAccount: Action[AnyContent] = Action.async { implicit request =>
-    withAuthenticatedAgent {
-      Future.successful {
-        request.session.get("arn").fold(sessionMissingRedirect()) { _ =>
-          Ok(html.link_account(linkAccountForm))
+    appConfig.autoMapAgentEnrolments match {
+      case true =>
+        withAuthenticatedAgent {
+          Future.successful {
+            request.session.get("arn").fold(sessionMissingRedirect()) { _ =>
+              Ok(html.link_account(linkAccountForm))
+            }
+          }
         }
-      }
+      case false => Future.successful(BadRequest)
```

Comment:
I would suggest to use 500 exception since the its us causing the exception, so that the end user will will get "there is something wrong" page.